linera-core/src/client.rs

```diff
@@ -1011,31 +1008,8 @@ where
         response.check(name)?;
         let mut certificates = Vec::new();
         let mut new_tracker = tracker;
-        for entry in response.info.requested_received_log {
-            let query = ChainInfoQuery::new(entry.chain_id)
-                .with_sent_certificate_hashes_in_range(BlockHeightRange::single(entry.height));
-            let local_response = node_client
-                .handle_chain_info_query(query.clone())
-                .await
-                .map_err(|error| NodeError::LocalNodeQuery {
-                    error: error.to_string(),
-                })?;
-            if !local_response
-                .info
-                .requested_sent_certificate_hashes
-                .is_empty()
-            {
-                new_tracker += 1;
-                continue;
-            }
-
-            let mut response = node.handle_chain_info_query(query).await?;
-            let Some(certificate_hash) = response.info.requested_sent_certificate_hashes.pop()
-            else {
-                break;
-            };
-
-            let certificate = node.download_certificate(certificate_hash).await?;
+        for hash in response.info.requested_received_log {
+            let certificate = node.download_certificate(hash).await?;
```

Comment:
Wouldn't it be more efficient to query the local node first? Since this is based on the validator-specific tracker, wouldn't we download each certificate from every validator?

Reply:
This is the current behavior. I'll look into that in the next PR

Reply:
Is it? Aren't we checking that by making the local node query you are removing?

Reply:
Oh you're right! My bad. Let me maintain this behavior then.

Reply:
@afck I updated the PR